### ISSUE TITLE
Release version does not match when installed

### ISSUE THREAD

**Issue by dbilgili** [mullvad/NSEventMonitor#25]
When I tried to install the package, I had the following error

```
node-pre-gyp WARN Using request for node-pre-gyp https download 
node-pre-gyp WARN Tried to download(404): https://github.com/mullvad/nseventmonitor/releases/download/0.0.19/nseventmonitor-v0.0.19-node-v83-darwin-x64.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for nseventmonitor@0.0.19 and node@14.15.1 (node-v83 ABI, unknown) (falling back to source compile with node-gyp) 
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'nan'
```

Tried installing `nan` globally, but that didn't help, then I installed it under `devDependencies` of my project.
Then I managed to install the `nseventmonitor` package with the following warnings

```
node-pre-gyp WARN Using request for node-pre-gyp https download 
node-pre-gyp WARN Tried to download(404): https://github.com/mullvad/nseventmonitor/releases/download/0.0.19/nseventmonitor-v0.0.19-node-v83-darwin-x64.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for nseventmonitor@0.0.19 and node@14.15.1 (node-v83 ABI, unknown) (falling back to source compile with node-gyp)
```

However, when I try to run my project, I get the following error.

```
Error: Cannot find module '../node_modules/nseventmonitor/lib/binding/Release/electron-v6.1-darwin-x64/nseventmonitor.node'
```

When I checked the installed package content and I figured that the `/Release` folder I got has the following folder structure, which does not match with what the package is looking for.

`node-v83-darwin-x64/nseventmonitor.node`

Any ideas on the possible fix for this problem?

**Comment by pronebird** [comment]
Hi,

It looks like we don't cater the pre-built binaries for your Node. 

It's possible to build the `nseventmonitor` from source and this is what `npm` should do by default.

```
npm install --build-from-source

> nseventmonitor@0.0.19 install /Users/pronebird/mullvad/mullvadvpn-app/gui/node_modules/nseventmonitor
> node-pre-gyp install --fallback-to-build

node-pre-gyp WARN Using needle for node-pre-gyp https download 
  CXX(target) Release/obj.target/nseventmonitor/src/addon.o
  CXX(target) Release/obj.target/nseventmonitor/src/NSEventMonitor.o
  CXX(target) Release/obj.target/nseventmonitor/src/NSEventMaskWrap.o
  CXX(target) Release/obj.target/nseventmonitor/src/NSEventWrap.o
  SOLINK_MODULE(target) Release/nseventmonitor.node
  COPY /Users/pronebird/mullvad/mullvadvpn-app/gui/node_modules/nseventmonitor/lib/binding/Release/node-v83-darwin-x64/nseventmonitor.node
  TOUCH Release/obj.target/action_after_build.stamp
```

However note that it built the binary for node-v83 where v83 refers to the version of ABI. I have `Node.js v14.6.0` in here. However Electron runs its own version of Node.js under the hood, so more importantly, we need to produce binaries for it.

In order to build the binary for Electron, you either can run a sketchy script like this on `postintall`:

```
ELECTRON_VERSION=`npx electron --version | sed  's/v//'`
ELECTRON_ABI=`npx electron --abi`

npm install --build-from-source --runtime=electron --target=$ELECTRON_VERSION --disturl=https://atom.io/download/atom-shell --abi=$ELECTRON_ABI
```

or what I would suggest is to use `electron-builder`  and add the following `postinstall` hook in your `package.json`:

```
electron-builder install-app-deps
```

If you ship for multiple platforms and include `nseventmonitor` as optional dependency, you may want to add an extra flag to tell `electron-builder` to ignore optional dependencies if they are missing:

```
ELECTRON_BUILDER_ALLOW_UNRESOLVED_DEPENDENCIES=true  electron-builder install-app-deps
```

Last but not least, if you opt-in for using `package.json` scripts, `cross-env` comes handy for aligning differences in how env variables are assed across platforms, so [the final script we use in Mullvad VPN](https://github.com/mullvad/mullvadvpn-app/blob/master/gui/package.json#L100) looks as following:

```
cross-env ELECTRON_BUILDER_ALLOW_UNRESOLVED_DEPENDENCIES=true electron-builder install-app-deps
```

I'll look into why `nan` not being pulled as it's clearly listed as `devDependency`. Maybe NPM works differently these days.

**Comment by pronebird** [comment]
@dbilgili Alright so the missing `nan` dependency issue is addressed in #27. Seems like we have to keep everything essential for building the module under `dependencies`. You can use `master` for now but there will be release later on.

**Comment by pronebird** [comment]
Please try the v0.0.20 release, npm should be able to build the module from source & you shouldn't need to have `nan` in your `devDependencies`.

I spoke to @raksooo and we may entirely abandon shipping the prebuilt binaries due to security concerns & general problem of serving builds for every single Node.js & Electron.

Also the build instructions for Electron were updated:
https://github.com/mullvad/NSEventMonitor#electron-support
